### PR TITLE
front: fix various bug with loading hider

### LINKED ISSUE #4759: Infra loading hider isn't properly aligned
### What happened?

This :
image

### What did you expect to happen?

The hider should align properly with other components.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to any scenario with an infra that needs to be loaded
2. Try to create a new train

### What operating system, browser and environment are you using?

- **Browser**: Chrome Version 115.0.5790.111
- **OS**: Windows 11
- **Env**: dev

### OSRD version (top right corner `Account` button > `Informations`)

Editoast 00a83cd2 Core 00a83cd2 API 00a83cd2 Front 00a83cd2

### LINKED ISSUE #5059: front: no message while infra is loading
### What happened?

When creating a scenario with an infra that hasn't been loaded yet, there isn't any message under the loader anymore.

image

### What did you expect to happen?

There should be an "Infra is loading" message under the loader

### How can we reproduce it (as minimally and precisely as possible)?

1. Create a scenario with an infra that hasn't been loaded yet

### What operating system, browser and environment are you using?

- **Browser**: Firefox v117
- **OS**: Ubuntu
- **Env**: Local / Dev

### OSRD version (top right corner `Account` button > `Informations`)

a797c532

### LINKED ISSUE #5078: Manage train schedule not available when infra is loading
### What happened?

When the infra is loading, managing train schedule (add or edit) is hidden by the waitingMessage component, preventing us from editing. 

### What did you expect to happen?

We should be able to edit train schedule when infra is loading.

### How can we reproduce it (as minimally and precisely as possible)?

Go to scenario
If infra isn't loaded, you can see the bug

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Nixos v22.11
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

87022614

### PR DESCRIPTION
**Pull request by Akctarus**
close #4759
close #5059 
close #5078

### CODE REVIEW

File: front/src/styles/scss/applications/operationalStudies/_scenario.scss [hunk 1]
```diff
@@ -48,6 +48,7 @@
     background-color: rgba(255, 255, 255, 0.5);
     z-index: 3;
     .scenario-loader-msg {
+      margin-top: 10px;
```
Comment (SharglutDev) [inline front/src/styles/scss/applications/operationalStudies/_scenario.scss]: You should do you margin in rem

**Review by SharglutDev — COMMENTED** [review]
Tested in local. Thanks for all these bug fixes !
Just one remark : it seems that when you remove all trains in timetable, it doesn't display "No Data to display"

File: front/src/common/Loader.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@ const LoaderSNCF: FC<{
 }> = ({ msg = '', position = 'center' }) => (
   <div className={`loader ${position}`}>
     <Spinner />
-    <div className="text-center mt-2">{msg}</div>
+    <div className="scenario-loader-msg">{msg}</div>
```
Comment (Math-R) [inline front/src/common/Loader.tsx]: This component is in common , you can't pass it contextual class name like this

**Review by Math-R — CHANGES_REQUESTED** [review]

File: front/src/common/Loader.tsx (hunk 2 above)
Comment (SharglutDev) [reply front/src/common/Loader.tsx]: The new WaitingMessage component should be in common too maybe ?

Comment (Math-R) [reply front/src/common/Loader.tsx]: Maybe but not in this form, if you want it to be a common component, you'll have to remove `<ScenarioLoader/>` from it and try to use more props

File: front/src/modules/scenario/components/ScenarioLoader.tsx [hunk 3]
```diff
@@ -8,8 +8,7 @@ type Props = {
 export default function ScenarioLoader({ msg }: Props) {
```
Comment (Math-R) [inline front/src/modules/scenario/components/ScenarioLoader.tsx]: Don't get the purpose of this component, as it duplicates, `<Loader/>` just to pass a single css class, I think it should be removed, and `<Loader/>` should be refactored, to handle the possibility to get custom class from the props

File: front/src/styles/scss/applications/operationalStudies/_scenario.scss (hunk 1 above)
Comment (Akctarus) [reply front/src/styles/scss/applications/operationalStudies/_scenario.scss]: thanks for the suggestion, fixed :)

File: front/src/modules/scenario/components/ScenarioLoader.tsx (hunk 3 above)
Comment (Akctarus) [reply front/src/modules/scenario/components/ScenarioLoader.tsx]: Done !

File: front/src/common/Loader.tsx (hunk 2 above)
Comment (Akctarus) [reply front/src/common/Loader.tsx]: Done :)

**Review by Math-R — APPROVED** [review]

**Pull request merged**